services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -46,7 +47,28 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
   override def getFileTree(project: Project, version: Option[Version])(
     implicit ec: ExecutionContext
   ): AsyncResult[List[String]] = ???
+
   override def getFile(project: Project, path: Path, version: Option[Version])(
     implicit ec: ExecutionContext
-  ): AsyncResult[String] = ???
+  ): AsyncResult[String] = {
+    val projectId: String = project.projectId.value
+    val filePath: String = URLEncoder.encode(path.toString, "UTF-8")
+    val fileVersion: String = version.map(el => el.value).getOrElse("master")
+
+    val responseFuture = httpClient.get(
+      s"${config.url}/projects/$projectId/repository/files/$filePath/raw",
+      Map("ref" -> fileVersion),
+      config.token
+    )
+
+    responseFuture
+      .map(resp => {
+        if (resp.status == StatusCodes.OK.intValue) {
+          Right(resp.body)
```

Comment:
Your method should return a List of strings not a body?

Reply:
We agreed that this line is right.